runtime/compiler/runtime/JITServerROMClassHash.cpp

```diff
@@ -84,8 +84,20 @@ JITServerROMClassHash::toString(char *buffer, size_t size) const
    TR_ASSERT(size > sizeof(_data) * 2, "Buffer too small");
 
    char *s = buffer;
+   size_t len = size;
    for (size_t i = 0; i < sizeof(_data); ++i)
-      s += sprintf(s, "%02x", ((uint8_t *)_data)[i]);
+      {
+      size_t amt = snprintf(s, len, "%02x", ((uint8_t *)_data)[i]);
+      if (amt <= len)
+         {
+         s += amt;
+         len -= amt;
+         }
+      else
+         {
+         len = 0;
+         }
+      }
```

Comment:
`snprintf()` returns an `int` (not `size_t`) which can be negative, in which case this will go wrong rather badly.
I'll make another pass through this to see whether that issue is repeated elsewhere.

Reply:
The body of the loop should be:
```c
      int amt = snprintf(s, len, "%02x", ((uint8_t *)_data)[i]);
      if ((0 < amt) && ((size_t)amt <= len))
         {
         s += amt;
         len -= amt;
         }
      else
         {
         break;
         }
```
Notice the `break` statement instead of clearing `len`: there's no point continuing in that case.